### DISCUSSION TITLE
Simulation "explodes" just by changing the size of the tank

### DISCUSSION THREAD

**Discussion by ItsIronOxide** [InteractiveComputerGraphics/SPlisHSPlasH#180] [Q&A]
Hi, I'm a master's student trying to use the SPlisHSPlasH open source code for my master's thesis. I am starting with a simple 3D dam break example similar to the one in [this video](https://www.youtube.com/watch?v=nKxsk00qB9o).

What I don't understand is why changing the length of the tank, to allow the water to flow further without hitting the opposite wall, would cause the simulation to "explode" with particles going crazy. After reading a lot in the literature about SPH and its boundary handling methods, I still haven't figured out why this happens.

Below are three scene files I used to generate this simulation. You'll find that the files are identical, except the length of the tank (in the `scale` field of `RigidBodies`). The simulation runs smoothly for a tank length of  0.84 m, explodes for 3.6 m, but works just fine again for 4 m.

I'd really appreciate some feedback about why this weird behaviour is happening, as I've been trying to find some answers for months now.

[dam_break_tank_length.zip](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/files/8762371/dam_break_tank_length.zip)

(Note: If some of the values in the scene JSON files seem awkward, that's because I'm generating them automatically with a python code I wrote, not writing them by hand. That's besides the point because the question here is: why does the simulation explode for specific tank sizes?)

**Comment by janbender** [comment]
I just tested you examples and found out that at the beginning the particles are not in the boundary object. So you cannot expect a stable simulation. 

A general hint: SPH tends to get chaotic motion if you do not have any viscosity in your simulation. Your visco coeff seems to be rather small. So you should at least use XSPH which is typically used in SPH to smooth the velocity field and to avoid chaotic motions. 

This SPH tutorial might be useful for you: 
https://interactivecomputergraphics.github.io/SPH-Tutorial/

**Reply by ItsIronOxide** [reply]
Thank you so much for your reply. Regarding the fact of the particles not being inside the boundary object, I'm actually using a different unit box that has the vertex at the origin, so it occupies the region `0<=x<=1, 0<=y<=1, 0<=z<=1`. By default, SPlisHSPlasH uses a unit box whose center is at the origin and ocuppies the region `-0.5<=x<=0.5, -0.5<=y<=0.5, -0.5<=z<=0.5`. I did this to avoid awkward tranlations in the coordinate system.

The unit box file I use is this one: [unit_box.zip](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/files/8762638/unit_box.zip)

So the particle placement in the beginning shouldn't be a problem (or is this new unit box file problematic?). Could you run the simulations with this unit box?

Regarding the viscosity, I'm using the kinematic viscosity of water, and it works sometimes. Other times it gets chaotic just by changing the size of the box.

Thank you so much for your hints. I have been stuck in this for months.

**Comment by janbender** [comment]
Okay, now the particles are exactly in the geometry. However, you are using an implicit boundary representation with a very low resolution. So either you increase the resolution of the boundary representation (adding the key "mapResolution" to the rigid body) or you use a particle-based boundary representation (by adding the key "boundaryHandlingMethod": 0 to the "Configuration"). 

Hope that helps!

**Reply by ItsIronOxide** [reply]
Thank you so much, Mr. Bender. Your help has been incredibly useful for my master's thesis. I hope to do great science with the help of SPlisHSPlasH.
